BE: reset dynamo before each test in test_module.py

shunting314 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #131397
* __->__ #131372
* #131388
* #131551

https://github.com/pytorch/pytorch/pull/126586 tried to reset dynamo before each unit test. That PR get reverted a couple of times because we see post-land test failures that we don't see before merge. This PR only reset dynamo before each tests in `test_module.py` to make it easier to land.

Eventually after we reset dynamo in each individual test files, we can move the change to the base class (TestCase) and remove the change in individual test files.

shunting314 committed:
BE: reset dynamo before each test in test_module.py

[ghstack-poisoned]

shunting314 committed:
Update on "BE: reset dynamo before each test in test_module.py"




[ghstack-poisoned]

zou3519 commented on test/test_modules.py:
```diff
@@ -25,6 +25,13 @@ class TestModule(TestCase):
     rel_tol = 1e-5
 
+    def setUp(self):
+        TestCase.setUp(self)
```
nit: super().setUp ?

shunting314 replied:
That's the first thing I tried :) But it fail with error:
```
TypeError: super(type, obj): obj must be an instance or subtype of type
```
I think it's due to the fact that we call instantiate_device_type_tests to create device special class'es

zou3519 replied:
Fair enough

zou3519 reviewed (changes requested):
Sorry, I changed my assessment: see the comments in https://github.com/pytorch/pytorch/pull/131388

shunting314 committed:
Update on "BE: reset dynamo before each test in test_module.py"


https://github.com/pytorch/pytorch/pull/126586 tried to reset dynamo before each unit test. That PR get reverted a couple of times because we see post-land test failures that we don't see before merge. This PR only reset dynamo before each tests in `test_module.py` to make it easier to land.

Eventually after we reset dynamo in each individual test files, we can move the change to the base class (TestCase) and remove the change in individual test files.


[ghstack-poisoned]

shunting314 committed:
Update on "BE: reset dynamo before each test in test_module.py"


https://github.com/pytorch/pytorch/pull/126586 tried to reset dynamo before each unit test. That PR get reverted a couple of times because we see post-land test failures that we don't see before merge. This PR only reset dynamo before each tests in `test_module.py` to make it easier to land.

Eventually after we reset dynamo in each individual test files, we can move the change to the base class (TestCase) and remove the change in individual test files.


[ghstack-poisoned]

kit1980 commented:
This broke https://github.com/pytorch/pytorch/actions/runs/10149118852/job/28065175173

kit1980 commented:
@pytorchbot revert -m "Broke test_modules.py::TestModuleCUDA::test_cpu_gpu_parity_nn_CTCLoss_cuda_float32 [GH job link](https://github.com/pytorch/pytorch/actions/runs/10149118852/job/28065175173) [HUD commit link](https://hud.pytorch.org/pytorch/pytorch/commit/ca8153ae6758fbf33cc767cfd0cb384b87b8d3ca)" -c nosignal

zou3519 commented:
@kit1980 do you know why there was no signal on this PR?

zou3519 commented:
> @kit1980 do you know why there was no signal on this PR?

Oh I see, ciflow/inductor

shunting314 commented:
I'm debugging the failure for CTCLoss. So far I found on CPU, the gradient for input is None while on GPU it's not. That cause the failure.

Right now CTCLoss is an instance of a parametrized test. I'll try to have an repro without parametrization. That may make the repro and fix easier.
